lib/player_input.rb

```diff
@@ -0,0 +1,51 @@
+class PlayerInput
+  attr_reader :input, :output, :validated_coordinates, :entered_coordinates
+
+  def initialize(input:, output:)
+    @input = input
+    @output = output
+  end
+
+  def coordinates
+    until validated_coordinates
+      get_and_validate_input
+    end
+    validated_coordinates
+  end
+
+  private
+
+  def get_and_validate_input
+    print_prompt
+    get_input
+    unless validate_input
+      print_error
+    end
+  end
+
+  def get_input
+    @entered_coordinates = input.gets.chomp
```

Comment:
I've found it challenging to model the loop of getting and validating input. Capturing some state here and in the method below feels _wrong_, but I can't put into words why. I tried pulling out another class but, but that felt worse somehow. I feel like there's something I'm missing or I'm overcomplicating things. Any guidance?